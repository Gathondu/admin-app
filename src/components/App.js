import React, { Component } from 'react';
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {BrowserRouter, Route, Switch} from "react-router-dom";
import HomePage from "./home/HomePage";
import AboutPage from "./about/AboutPage";
import Header from "./common/Header";
import Error from "./common/Error";
import CoursesPage from "./course/CoursesPage";
import ManageCoursePage from './course/ManageCoursePage';
import {bindActionCreators} from "redux";
import * as CourseActions from '../actions/course';
import AuthorsPage from "./author/AuthorsPage";
import ManageAuthorPage from "./author/ManageAuthorPage";
// import toastr from 'toastr';

export class App extends Component {
    static propTypes = {
        courses: PropTypes.array.isRequired,
        loading: PropTypes.bool.isRequired,
        totalCourses: PropTypes.number.isRequired,
        totalAuthors: PropTypes.number.isRequired
    };

    render() {
        const { courses, loading, dispatch, totalCourses, totalAuthors} = this.props;
        const deleteCourse = bindActionCreators(CourseActions.deleteCourse, dispatch);
        // const handleDeleteAuthor = authorId => {
        //     const authorToDelete = courses.filter(course => course.authorId === authorId);
        //     (authorToDelete.length > 0)
        //         ? toastr.error('Cannot' +
        //         '' +
        //         ' delete Author.They have a registered course.')
        //         : deleteAuthor(authorId)
        // };
        return (
            <BrowserRouter>
                <div className="container">
                    <Header loading={loading} totalCourses={totalCourses} totalAuthors={totalAuthors}/>
                    <Switch>
                        <Route exact path="/" render={() => <HomePage title="Administration" />}/>
                        <Route exact path="/authors" component={AuthorsPage}/>
                        <Route exact path="/author" render={(props) => <ManageAuthorPage {...props}/>}/>
                        <Route exact path="/author/:id" render={(props) => <ManageAuthorPage {...props}/>}/>
                        <Route exact path="/courses" render={(props) =>
                            <CoursesPage {...props}
                                         courses={courses}
                                         deleteCourse={deleteCourse}
                                         loading={loading} />}/>
                        <Route exact path="/course" render={(props) => <ManageCoursePage {...props}/>}/>
                        <Route exact path="/course/:id" render={(props) => <ManageCoursePage {...props}/>}/>
                        <Route eaxct path="/about" render={() => <AboutPage title="About"/>}/>
                        <Route component={Error}/>
                    </Switch>
                </div>
            </BrowserRouter>
        );
    }
}
export const mapStateToProps = state => (
    {
        courses: state.courses,
        authors: state.authors,
        loading: state.ajaxCallsInProgress > 0,
        totalCourses: state.totalCourses,
        totalAuthors: state.totalAuthors
    }
);

export default connect(mapStateToProps)(App);
