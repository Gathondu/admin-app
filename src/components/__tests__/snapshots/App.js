import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {App} from '../../App';
describe('App Container Component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<App totalCourses={4} totalAuthors={3} loading={false} courses={[]}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});