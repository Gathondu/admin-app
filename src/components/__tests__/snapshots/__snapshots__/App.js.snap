// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container Component should render without crashing 1`] = `
<BrowserRouter>
  <div
    className="container"
  >
    <Header
      loading={false}
      totalAuthors={3}
      totalCourses={4}
    />
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/authors"
      />
      <Route
        exact={true}
        path="/author"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/author/:id"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/courses"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/course"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/course/:id"
        render={[Function]}
      />
      <Route
        eaxct={true}
        path="/about"
        render={[Function]}
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
